recipes/opencv/4.x/conanfile.py

```diff
@@ -34,6 +34,7 @@ class OpenCVConan(ConanFile):
         "with_cublas": [True, False],
         "with_cufft": [True, False],
         "with_v4l": [True, False],
+        "with_neon": [None, True, False],
```

Comment:
I don't like "auto" options values.

You could set this value to True by default, and delete it in config_options if those conditions are not met: https://github.com/opencv/opencv/blob/c0f63eb21f044415bf7a21cb98fa6f4ae2586e1b/CMakeLists.txt#L494
(it's also more complex to implement, but package id are neat, and build predictible).

Reply:
But setting it to true by default does not replicate their logic: `(NEON OR ANDROID_ARM_NEON OR AARCH64)`

I can adapt the default logic, AND delete the option when `(CV_GCC OR CV_CLANG) AND (ARM OR AARCH64 OR IOS)` is not the case.

If OpenCV changes its logic, then we are out of synch again. Does it make sense to have such a tight coupling? I see this as an advanced, optional flag, most people won't ever need to touch.

Reply:
The logic is simple: neon enabled by default for archs supporting neon. By deleting this option if arch doesn't support neon, and injecting `self.options.get_safe("with_neon", False)`, you preserve upstream logic, and avoid auto value.

Reply:
@SSE4 since you approved, any opinion on @SpaceIm 's remark?

Reply:
I have no objections to adding an option. this is clearly an improvement over current logic (which already auto-detects).

Reply:
I would prefer @SpaceIm suggestion, not because you option is not clear, but because we already have a pattern for most of recipes. Also, "auto" looks easier for noobs which doesn't know about OpenCV.

Reply:
okidoki, will cook something up

Reply:
@SpaceIm @uilianries done, I took the logik from
https://github.com/conan-io/conan-center-index/blob/master/recipes/gf-complete/all/conanfile.py#L48-L49